core/controllers/improvements.py

```diff
@@ -0,0 +1,111 @@
+from core.platform import models
+import feconf
+
+(improvements_models,) = (
+    models.Registry.import_models([models.NAMES.improvements]))
+
+
+class ExplorationImprovementsHandler(base.BaseHandler):
+    """Handles operations related to managing exploration improvement tasks.
+
+    NOTE: Only exploration creators can interface with tasks.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_edit_exploration
+    def get(self, exploration_id):
+        open_tasks, resolved_task_types_by_state_name = (
+            improvements_services.fetch_exploration_tasks(
+                exp_fetchers.get_exploration_by_id(exploration_id)))
+        self.render_json({
+            'open_tasks': [t.to_dict() for t in open_tasks],
+            'resolved_task_types_by_state_name': (
+                resolved_task_types_by_state_name),
+        })
+
+    @acl_decorators.can_edit_exploration
+    def post(self, exploration_id):
+        task_entries = self.payload.get('task_entries', None)
+        if task_entries is None:
+            raise self.InvalidInputException('No task_entries provided')
+        task_entries_to_put = []
+        for task_entry in task_entries:
+            entity_version = task_entry.get('entity_version', None)
+            if entity_version is None:
+                raise self.InvalidInputException('No entity_version provided')
+            task_type = task_entry.get('task_type', None)
+            if task_type is None:
+                raise self.InvalidInputException('No task_type provided')
+            target_id = task_entry.get('target_id', None)
+            if target_id is None:
+                raise self.InvalidInputException('No target_id provided')
+            status = task_entry.get('status', None)
+            if status is None:
+                raise self.InvalidInputException('No status provided')
+            # The issue_description is allowed to be None.
+            issue_description = task_entry.get('issue_description', None)
+            task_entries_to_put.append(
+                improvements_domain.TaskEntry(
+                    improvements_models.TASK_ENTITY_TYPE_EXPLORATION,
+                    exploration_id,
+                    entity_version,
+                    task_type,
+                    improvements_models.TASK_TARGET_TYPE_STATE,
+                    target_id,
+                    issue_description,
+                    status,
+                    self.user_id,
+                    datetime.datetime.utcnow()))
+        improvements_services.put_tasks(task_entries_to_put)
+        self.render_json({})
+
+
+class ExplorationImprovementsHistoryHandler(base.BaseHandler):
+    """Handles fetching the history of resolved exploration tasks.
+
+    NOTE: Only exploration creators can interface with tasks.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_edit_exploration
```

Comment:
The system can be extended to support voiceovers eventually, yes, but for now these tasks are aimed only at editors because they show items that are only designed to be consumed by them. The tab will be invisible for non-editors as well.

Reply:
I see, thanks for the explanation! :)